tests/manifold/transfinite_manifold_11.cc

```diff
@@ -0,0 +1,155 @@
+    Assert(point[0] >= 0., ExcInternalError());
+    return polar_manifold.push_forward(point);
+  }
+
+  std::unique_ptr<Manifold<dim, dim>>
+  clone() const final override
+  {
+    return std::make_unique<GradingManifold<dim>>(center, grading, epsilon);
+  }
+
+private:
+  const Point<dim> center;
+  const double     grading;
+  const double     epsilon;
+
+  PolarManifold<dim> polar_manifold;
+};
+
+
+int
+main()
+{
+  initlog();
+  deallog << std::setprecision(9);
+
+  const std::vector<Point<2>> vertices{
+    {0.5 * 2.5, -std::sqrt(3.) / 2. * 2.5}, // 0
+    {-0.9 + 0.7, -0.025},                   // 1
+    {-0.9 + 0.7, 0.025},                    // 2
+    {0.5 * 2.5, std::sqrt(3.) / 2. * 2.5},  // 3
+    {2.5, -0.5 * 2.5},                      // 4
+    {2.5, -0.416667},                       // 5
+    {2.5, 0.416667},                        // 6
+    {2.5, 0.5 * 2.5},                       // 7
+  };
+
+  std::vector<CellData<2>> cells(3);
+  cells[0].vertices = {0, 4, 1, 5};
+  cells[1].vertices = {1, 5, 2, 6};
+  cells[2].vertices = {2, 6, 3, 7};
+
+  Triangulation<2> triangulation;
+  triangulation.create_triangulation(vertices, cells, SubCellData());
+
+  {
+    triangulation.set_all_manifold_ids(3);
+    const auto center_cell  = ++triangulation.begin_active();
+    const auto lower_radial = center_cell->face(2);
+    const auto upper_radial = center_cell->face(3);
+    lower_radial->set_manifold_id(1);
+    upper_radial->set_manifold_id(2);
+  }
+
+  GradingManifold<2> lower_radial{Point<2>{-0.9 + 0.7, -0.025}, 3.0, 0.01};
+  triangulation.set_manifold(1, lower_radial);
+
+  GradingManifold<2> upper_radial{Point<2>{-0.9 + 0.7, 0.025}, 3.0, 0.01};
+  triangulation.set_manifold(2, upper_radial);
+
+  TransfiniteInterpolationManifold<2> transfinite;
+  transfinite.initialize(triangulation);
+  triangulation.set_manifold(3, transfinite);
+
+  /*
+   * triangulation.refine_global(8) fails with:
+   *
+   *   An error occurred in line [...] in function
+   *       [...]compute_chart_points(const ArrayView<const Point<spacedim>> &,
+   *   ArrayView<Point<dim>>) The violated condition was: false Additional
+   *   information: Looking at cell 1_0: with vertices: -0.2 -0.025    2.5
+   *   -0.416667    -0.2 0.025    2.5 0.416667 Transformation to chart
+   *   coordinates: -0.1271 -0.0177875 -> 0.1875 0.25 -0.1271 -0.015564 -> 20 20
+   *   Looking at cell 0_0: with vertices:
+   *   1.25 -2.16506    2.5 -1.25    -0.2 -0.025    2.5 -0.416667
+   *   Transformation to chart coordinates:
+   *   -0.1271 -0.0177875 -> 0.211251 1.01047
+   *   -0.1271 -0.015564 -> 0.214091 1.01181
+   *   Looking at cell 2_0: with vertices:
+   *   -0.2 0.025    2.5 0.416667    1.25 2.16506    2.5 1.25
+   *   Transformation to chart coordinates:
+   *   -0.1271 -0.0177875 -> 0.253794 -0.0323121
+   *   -0.1271 -0.015564 -> 0.251301 -0.0309146
+   *
+   * The following is a simplified version that fails with a sligthly
+   * different error message.
```

Comment:
@kronbichler This took a while to extract. I have simplified the test from globally refining 8 times to a single `get_new_point` invocation. The manifold then fails with a slightly different errror path. In any case, #11131 fixes the problem.

Reply:
Yes, this looks exactly as I would have done it, nice test!